### ISSUE TITLE
Grid

### ISSUE THREAD

**Issue by vanillajonathan** [gaphor/gaphor#798]
I would like an option to enable displaying a grid overlay.

The grid overlay would be useful in helping the user placing elements. There could also be a snap-to-grid option where if you drag elements they automatically snap to grid if moved like maybe 10 pixels close.

Grids are illustrated in the GIMP documentation.
https://docs.gimp.org/2.10/en/gimp-concepts-image-grid-and-guides.html

**Comment by danyeaw** [comment]
@vanillajonathan I use other modeling tools, and the grid is always the first thing I turn off! :smile: 

I don't think this would be a priority for us to implement right now, but we are open to contributions if this is important for you.

**Comment by vanillajonathan** [comment]
While for drawing free hand things like cartoons the grid isn't useful, but I believe that for technical drawing where you want elements aligned, then the grid is very useful, especially together with a snap-to-grid functionality.

**Comment by JennyPre** [comment]
I also think that some shortcut or button to evenly distribute items evenly would be very useful. Kind of like it is implemented in [Inkscape](https://inkscape-manuals.readthedocs.io/en/latest/align-and-distribute.html).

**Comment by h8672** [comment]
It would help to have a set accuracy for item to snap on. Like 2 mm and it calculates moused drag XY position always to snap on those... 16.9746 cm X snaps to 16.96 cm or 16.98 cm (because I don't need approximate, just accurate snaps).

I noticed under Interactions [Interaction] using [Execution specification] on [Lifeline] lines really like to jump a lot with their current snapping with near edges. I think it warps when mouse slightly trembles.

So... option to enable pixel / cm snap for dragged items at valid distances during drag events?
Grid would be just visual representation for measure and it could be more loose if people want with own adjustment.
Putting same value for them would be like it would snap on grid.

**Comment by gustycroc** [comment]
I agree completely, a toggleable snap-to-grid feature would be wonderful, not only for placing items, but also resizing them (so all dimensions would be in multiples of a grid unit). It's a feature I find very comfortable for bigger diagrams with many components.
Likewise, it would be nice to be able to toggle the relative snap/alignment between items on and off. It's nifty, but as soon as I have more than two items that should be aligned vertically or horizontally, Gaphor seems to struggle with which reference my current item should snap to. Thank you!
